File: 0257-binary-tree-paths/0257-binary-tree-paths.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    void helper(List<String> ans, TreeNode root, String a)
    {
        if(root == null)
            return;
        a += String.valueOf(root.val);
        if(root.left == null && root.right == null)
        {
            ans.add(a);
            return;
        }
        helper(ans, root.left, a+"->");
        helper(ans, root.right, a+"->");
    }
    public List<String> binaryTreePaths(TreeNode root) {
        List<String> ans = new ArrayList<>();
        helper(ans, root, "");
        return ans;
    }
}

File: 0257-binary-tree-paths/0257-binary-tree-paths.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
public:
    void helper(vector<string>& ans, TreeNode* root, string a)
    {
        if(root == NULL)
            return;
        if(! a.empty())
            a += "->";
        a += to_string(root->val);
        if(root->left == NULL && root->right == NULL)
        {
            ans.push_back(a);
            return;
        }
        helper(ans, root->left, a);
        helper(ans, root->right, a);
    }
    vector<string> binaryTreePaths(TreeNode* root) {
        vector<string> ans;
        helper(ans, root, "");
        return ans;
    }
};